docs/doc-iothub-storage-private.md

```diff
@@ -0,0 +1,150 @@
+
+The file upload functionality provided by Azure IoT Hub serves as a secure bridge to the Azure Storage account. This eliminates the need for IoT client devices to rely on external custom services for secure connections to the Storage account. Instead, Azure IoT Hub exposes APIs to the client, handling this aspect seamlessly. For a comprehensive understanding of this feature, refer to the official  [Azure IoT Hub file upload overview](https://learn.microsoft.com/azure/iot-hub/iot-hub-devguide-file-upload).
+
+This document offers guidance on securing Azure Storage account through a private endpoint while still allowing IoT Hub to interact with it, as well as allowing client IoT devices to reach the account through a gateway (Application Gateway and optionally Azure Firewall). With this approach the Storage account no longer allows public Internet access, which is a common requirement in enterprise deployments.
+
+This sequence diagram illustrates the process of a client device uploading a file to Azure Storage via Azure IoT Hub. Here's a step-by-step breakdown:
+
+```mermaid
+sequenceDiagram
+    participant AzureDevice as Client Device
+    participant AzureIoTHub as Azure IoT Hub
+    participant AzureStorage as Azure Storage
+
+    AzureDevice->>AzureIoTHub: Requests a SAS URI for file upload
+    AzureIoTHub->>AzureStorage: Brokers the creation of the SAS URI
+    AzureStorage-->>AzureIoTHub: Returns the SAS URI
+    AzureIoTHub-->>AzureDevice: Sends SAS URI
+    AzureDevice->>AzureStorage: Device uses Blob Storage SDK to talk to Azure Storage with SAS URI
+    AzureDevice->>AzureIoTHub: Device informs IoT Hub of file upload complete
+```
+
+## Moving towards an Azure Storage account without public Internet access
+
+In regular deployments, because the Azure IoT client device needs to talk directly to the Storage account to upload the file, meaning this Storage account needs to allow incoming Internet traffic. Disabling Internet traffic would automatically block any client IoT devices from uploading files.
+
+Customers with more stringent networking requirements may want to leverage a common practice of making the Azure Storage account accessible only from within a private network and inspect any traffic through a firewall or gateway. Often the implementation will be in the form of a **Hub-Spoke network topology**. The concepts for this architecture is discussed in the Azure architecture center article: Hub-spoke network topology in [Azure Architecture Center | Microsoft Learn](https://learn.microsoft.com/azure/architecture/networking/architecture/hub-spoke?tabs=cli).
+
+In this article, we outline a strategy to enhance the security of Azure IoT Hub's file upload feature. This approach blocks direct internet traffic to the Storage account, permitting only the traffic routed through the inbound Application Gateway. This setup also enables traffic inspection via Azure Firewall for added security.
```

Comment:
to align with the rest of the doc you should write "internet" with a starting capital letter

Reply:
Good catch :)